broken mobile Solidjs template

```
cargo create-tauri-app --alpha
✔ Project name · charge-spy
✔ Choose which language to use for your frontend · TypeScript / JavaScript - (pnpm, yarn, npm)
✔ Choose your package manager · pnpm
✔ Choose your UI template · Solid - (https://www.solidjs.com/)
✔ Choose your UI flavor · TypeScript
✔ Would you like to setup the project for mobile as well? · yes

```


```
pnpm tauri dev        

> charge-spy@0.0.0 tauri C:\Users\acoop\Documents\GitHub\ChargeSpy
> tauri "dev"

     Running BeforeDevCommand (`pnpm dev`)

> charge-spy@0.0.0 dev C:\Users\acoop\Documents\GitHub\ChargeSpy
> vite

X [ERROR] Top-level await is currently not supported with the "cjs" output format

    vite.config.ts:25:18:
      25 │             host: await internalIpV4(),
         ╵                   ~~~~~

failed to load config from C:\Users\acoop\Documents\GitHub\ChargeSpy\vite.config.ts
ild\lib\main.js:756:9)
    at Socket.readFromStdout (C:\Users\acoop\Documents\GitHub\ChargeSpy\node_modules\.pnpm\esbuild@0.16.17\node_modules\esbuild\lib\main.js:677:7)
    at Socket.emit (node:events:513:28)    at addChunk (node:internal/streams/readable:324:12)
 ELIFECYCLE  Command failed with exit code 1.       Error The "beforeDevCommand" terminated with a non-zero status code.
 ELIFECYCLE  Command failed with exit code 1.PS C:\Users\acoop\Documents\GitHub\ChargeSpy>
```

--- a/.changes/solid-js-type-module.md
+++ b/.changes/solid-js-type-module.md
@@ -1,0 +1,6 @@
+---
+"create-tauri-app": "patch"
+"create-tauri-app-js": "patch"
+---
+
+Add missing `"type": "module"` to `package.json` in SolidJs templates.

--- a/packages/cli/fragments/fragment-react-ts/%(mobile)%vite.config.ts
+++ b/packages/cli/fragments/fragment-react-ts/%(mobile)%vite.config.ts
@@ -2,8 +2,12 @@ import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [react()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -11,21 +15,15 @@ export default defineConfig({
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -39,4 +37,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-react-ts/%(stable-alpha)%vite.config.ts
+++ b/packages/cli/fragments/fragment-react-ts/%(stable-alpha)%vite.config.ts
@@ -1,8 +1,12 @@
 import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [react()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -24,4 +28,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-react/%(mobile)%vite.config.js
+++ b/packages/cli/fragments/fragment-react/%(mobile)%vite.config.js
@@ -2,8 +2,12 @@ import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [react()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -11,21 +15,15 @@ export default defineConfig({
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -39,4 +37,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-react/%(stable-alpha)%vite.config.js
+++ b/packages/cli/fragments/fragment-react/%(stable-alpha)%vite.config.js
@@ -1,8 +1,12 @@
 import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [react()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -24,4 +28,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-solid-ts/%(alpha)%package.json
+++ b/packages/cli/fragments/fragment-solid-ts/%(alpha)%package.json
@@ -2,6 +2,7 @@
   "name": "{{package_name}}",
   "version": "0.0.0",
   "description": "",
+  "type": "module",
   "scripts": {
     "start": "vite",
     "dev": "vite",

--- a/packages/cli/fragments/fragment-solid-ts/%(mobile)%package.json
+++ b/packages/cli/fragments/fragment-solid-ts/%(mobile)%package.json
@@ -2,6 +2,7 @@
   "name": "{{package_name}}",
   "version": "0.0.0",
   "description": "",
+  "type": "module",
   "scripts": {
     "start": "vite",
     "dev": "vite",

--- a/packages/cli/fragments/fragment-solid-ts/%(mobile)%vite.config.ts
+++ b/packages/cli/fragments/fragment-solid-ts/%(mobile)%vite.config.ts
@@ -2,8 +2,12 @@ import { defineConfig } from "vite";
 import solidPlugin from "vite-plugin-solid";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [solidPlugin()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -11,21 +15,15 @@ export default defineConfig({
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -39,4 +37,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-solid-ts/%(stable-alpha)%vite.config.ts
+++ b/packages/cli/fragments/fragment-solid-ts/%(stable-alpha)%vite.config.ts
@@ -1,8 +1,12 @@
 import { defineConfig } from "vite";
 import solidPlugin from "vite-plugin-solid";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [solidPlugin()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -24,4 +28,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-solid/%(alpha)%package.json
+++ b/packages/cli/fragments/fragment-solid/%(alpha)%package.json
@@ -2,6 +2,7 @@
   "name": "{{package_name}}",
   "version": "0.0.0",
   "description": "",
+  "type": "module",
   "scripts": {
     "start": "vite",
     "dev": "vite",

--- a/packages/cli/fragments/fragment-solid/%(mobile)%package.json
+++ b/packages/cli/fragments/fragment-solid/%(mobile)%package.json
@@ -2,6 +2,7 @@
   "name": "{{package_name}}",
   "version": "0.0.0",
   "description": "",
+  "type": "module",
   "scripts": {
     "start": "vite",
     "dev": "vite",

--- a/packages/cli/fragments/fragment-solid/%(mobile)%vite.config.js
+++ b/packages/cli/fragments/fragment-solid/%(mobile)%vite.config.js
@@ -2,8 +2,12 @@ import { defineConfig } from "vite";
 import solidPlugin from "vite-plugin-solid";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [solidPlugin()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -11,21 +15,15 @@ export default defineConfig({
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -39,4 +37,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-solid/%(stable-alpha)%vite.config.js
+++ b/packages/cli/fragments/fragment-solid/%(stable-alpha)%vite.config.js
@@ -1,8 +1,12 @@
 import { defineConfig } from "vite";
 import solidPlugin from "vite-plugin-solid";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [solidPlugin()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -24,4 +28,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-svelte-ts/%(mobile)%vite.config.ts
+++ b/packages/cli/fragments/fragment-svelte-ts/%(mobile)%vite.config.ts
@@ -3,8 +3,12 @@ import { svelte } from "@sveltejs/vite-plugin-svelte";
 import sveltePreprocess from "svelte-preprocess";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [
     svelte({
       preprocess: [
@@ -20,21 +24,15 @@ export default defineConfig({
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -48,4 +46,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-svelte-ts/%(stable-alpha)%vite.config.ts
+++ b/packages/cli/fragments/fragment-svelte-ts/%(stable-alpha)%vite.config.ts
@@ -2,8 +2,12 @@ import { defineConfig } from "vite";
 import { svelte } from "@sveltejs/vite-plugin-svelte";
 import sveltePreprocess from "svelte-preprocess";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [
     svelte({
       preprocess: [
@@ -33,4 +37,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-svelte/%(mobile)%vite.config.js
+++ b/packages/cli/fragments/fragment-svelte/%(mobile)%vite.config.js
@@ -2,8 +2,12 @@ import { defineConfig } from "vite";
 import { svelte } from "@sveltejs/vite-plugin-svelte";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [svelte()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -11,21 +15,15 @@ export default defineConfig({
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -39,4 +37,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-svelte/%(stable-alpha)%vite.config.js
+++ b/packages/cli/fragments/fragment-svelte/%(stable-alpha)%vite.config.js
@@ -1,8 +1,12 @@
 import { defineConfig } from "vite";
 import { svelte } from "@sveltejs/vite-plugin-svelte";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [svelte()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -24,4 +28,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-vanilla-ts/%(mobile)%vite.config.ts
+++ b/packages/cli/fragments/fragment-vanilla-ts/%(mobile)%vite.config.ts
@@ -1,28 +1,26 @@
 import { defineConfig } from "vite";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
   // prevent vite from obscuring rust errors
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -36,4 +34,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-vanilla-ts/%(stable-alpha)%vite.config.ts
+++ b/packages/cli/fragments/fragment-vanilla-ts/%(stable-alpha)%vite.config.ts
@@ -1,7 +1,11 @@
 import { defineConfig } from "vite";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
   // prevent vite from obscuring rust errors
   clearScreen: false,
@@ -21,4 +25,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-vue-ts/%(mobile)%vite.config.ts
+++ b/packages/cli/fragments/fragment-vue-ts/%(mobile)%vite.config.ts
@@ -2,8 +2,12 @@ import { defineConfig } from "vite";
 import vue from "@vitejs/plugin-vue";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [vue()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -11,21 +15,15 @@ export default defineConfig({
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -39,4 +37,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-vue-ts/%(stable-alpha)%vite.config.ts
+++ b/packages/cli/fragments/fragment-vue-ts/%(stable-alpha)%vite.config.ts
@@ -1,8 +1,12 @@
 import { defineConfig } from "vite";
 import vue from "@vitejs/plugin-vue";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [vue()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -24,4 +28,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-vue/%(mobile)%vite.config.js
+++ b/packages/cli/fragments/fragment-vue/%(mobile)%vite.config.js
@@ -2,8 +2,12 @@ import { defineConfig } from "vite";
 import vue from "@vitejs/plugin-vue";
 import { internalIpV4 } from "internal-ip";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [vue()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -11,21 +15,15 @@ export default defineConfig({
   clearScreen: false,
   // tauri expects a fixed port, fail if that port is not available
   server: {
-    host:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? "0.0.0.0"
-        : false,
+    host: mobile ? "0.0.0.0" : false,
     port: 1420,
-    hmr:
-      process.env.TAURI_PLATFORM === "android" ||
-      process.env.TAURI_PLATFORM === "ios"
-        ? {
-            protocol: "ws",
-            host: await internalIpV4(),
-            port: 1421,
-          }
-        : undefined,
+    hmr: mobile
+      ? {
+          protocol: "ws",
+          host: await internalIpV4(),
+          port: 1421,
+        }
+      : undefined,
     strictPort: true,
   },
   // to make use of `TAURI_DEBUG` and other env variables
@@ -39,4 +37,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

--- a/packages/cli/fragments/fragment-vue/%(stable-alpha)%vite.config.js
+++ b/packages/cli/fragments/fragment-vue/%(stable-alpha)%vite.config.js
@@ -1,8 +1,12 @@
 import { defineConfig } from "vite";
 import vue from "@vitejs/plugin-vue";
 
+const mobile =
+  process.env.TAURI_PLATFORM === "android" ||
+  process.env.TAURI_PLATFORM === "ios";
+
 // https://vitejs.dev/config/
-export default defineConfig({
+export default defineConfig(async () => ({
   plugins: [vue()],
 
   // Vite options tailored for Tauri development and only applied in `tauri dev` or `tauri build`
@@ -24,4 +28,4 @@ export default defineConfig({
     // produce sourcemaps for debug builds
     sourcemap: !!process.env.TAURI_DEBUG,
   },
-});
+}));

